src/parser/keyword.c

```diff
@@ -556,18 +598,55 @@ pt_find_keyword (const char *text)
       return NULL;
     }
 
-  if (strlen (text) >= MAX_KEYWORD_SIZE)
+  len = strlen (text);
+  if (len < keyword_min_len || len > keyword_max_len)
     {
       return NULL;
     }
 
+#if 1
+  /* Keywords are composed of ASCII characters(English letters, underbar).  */
+  unsigned char *p, *s;
+  s = (unsigned char *) dummy.keyword;
+  for (p = (unsigned char *) text; *p; p++, s++)
+    {
+      if (*p >= 0x80)
+	return NULL;
+
+      if (*p >= 'a' && *p <= 'z')
+	*s = *p + ('A' - 'a');
+      else if (*p < 0x80)
+	*s = *p;
```

Comment:
converting to uppercase already implemented at `char_toupper() in chartype.c`

Reply:
That function is slow.
Also, here we try to return NULL immediately if it is not an ASCII character.

Reply:
Returning NULL after checking a character is not related to the function, `char_toupper()`. It would better reuse it to remove duplicated code.

Reply:
OK,  I will fix it.